narwhal/executor/src/subscriber.rs

```diff
@@ -164,65 +164,68 @@ impl<Network: SubscriberNetwork> Subscriber<Network> {
     }
 }
 
+async fn run_notify<State: ExecutionState + Send + Sync + 'static>(
+    state: State,
+    mut tr_notify: metered_channel::Receiver<ConsensusOutput>,
+) {
+    loop {
+        while let Some(message) = tr_notify.recv().await {
+            let sub_dag = message.sub_dag.clone();
+            state.notify_commit_boundary(&sub_dag).await;
```

Comment:
I think at this point it is reasonable to remove `notify_commit_boundary` method from the ExecutorState.

In the ConsensusHandler you can call `Authority::notify_commit_boundary` at the end of the dag processing